ckks/params.go

```diff
@@ -311,7 +292,7 @@ func (p ParametersLiteral) RLWEParameters() rlwe.ParametersLiteral {
 		H:              p.H,
 		RingType:       p.RingType,
 		DefaultNTTFlag: DefaultNTTFlag,
-		DefaultScale:   rlwe.NewScale(p.DefaultScale),
+		DefaultScale:   rlwe.NewScale(math.Exp2(float64(p.LogScale))),
```

Comment:
Both `float64(p.LogScale)` and `math.Exp2(float64)` can return incorrect results silently. I was unable to find nice golang standard library stuff that `panic()`s instead of returning incorrect results silently, so if you know them, please use them. An alternative is to implement a function:
```
panic_float64(value: int64) {
  assert(value < (1 << 52))
  return float64(value)
}
```
and
```
fn panic_exp2(value: float64) {
  let result = math.Exp2(value)
  assert(!math.IsNan(result) and !math.IsInf(result) and result != 0.0)
  return result
}
```
(I don't know Go, but you get the picture).

Then do 
```
rlwe.NewScale(panic_exp2(panic_float64(p.LogScale))
```

While these edgecases are unlikely, if they were ever to happen, tracking down these bugs would be very hard.

Reply:
The default scaling factor cannot be given as a float anymore. It is now given in log2 as an int, which means that it will always be a power of two, and thus will always have an exact representation. As the default scaling factor is read only, it can never be modified, such edge cases can never happen.